lib/http.js

```diff
@@ -121,17 +121,17 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
     validate({query: 'bedrock-profile-http.account'}),
     asyncHandler(async (req, res) => {
       const {account: {id: accountId}} = (req.user || {});
-      const {account} = req.query;
+      const {profileAgentId} = req.params;
+      // this throws NotFoundError
+      const profileAgentRecord = await profileAgents.get({id: profileAgentId});
       // FIXME: Use http-sigs and zcaps for
       // TODO: Add permissions to allow access for admin
-      if(account !== accountId) {
+      if(profileAgentRecord.profileAgent.account !== accountId) {
```

Comment:
Do we know if we need to retrieve profile agents that have no account set? Or what about profile agents with `token` set, i.e., "integrations/applications"?

Reply:
This actually might be a new bug. You could previously get an integration provided the account parameter in the request matched the accountId of the authenticated account. So my apologies about this one and we do need to decide on how to handle integrations. One idea might be to get all profileAgents that belong to the profile the integration belongs to, make a list of account ids, and then if the authenticated account is in that list allow the integration to be returned.

Reply:
https://github.com/digitalbazaar/bedrock-profile-http/pull/32/commits/f6a7022772ba6d8cee1a482517949f56ee54a412

Added logic for this here. The rule being: if there is an associated account don't allow them to get a profileAgent unless the authenticated user is that account. Is they are requesting a intergration or unclaimed profile, only allow it if the authenticated account has a profileAgent that is part of the same profile. There is now a broken test that needs to be corrected.